src/porting.cpp

```diff
@@ -697,6 +699,20 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
 	return c;
 }
 
+bool openURL(std::string url)
+{
+	if (!std::regex_match(url, std::regex("^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$") )) {
```

Comment:
```
porting.cpp: In function ‘bool porting::openURL(std::__cxx11::string)’:
porting.cpp:704:40: warning: unknown escape sequence: '\/'
  if (!std::regex_match(url, std::regex("^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$") )) {
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
porting.cpp:704:40: warning: unknown escape sequence: '\/'
porting.cpp:704:40: warning: unknown escape sequence: '\d'
porting.cpp:704:40: warning: unknown escape sequence: '\.'
porting.cpp:704:40: warning: unknown escape sequence: '\.'
porting.cpp:704:40: warning: unknown escape sequence: '\.'
porting.cpp:704:40: warning: unknown escape sequence: '\/'
porting.cpp:704:40: warning: unknown escape sequence: '\w'
porting.cpp:704:40: warning: unknown escape sequence: '\.'
porting.cpp:704:40: warning: unknown escape sequence: '\/'
```

Reply:
The problem with `\d` is that it escapes `d`, and expects it to be an control ASCII character. Try `\\d` which then becomes the literal `\d` in the final string.
Possibly simpler URL check which isn't that precise but allows special characters for certain TLDs:
`^(https?://([^ /]+)\\.([a-z]{2,}/?` (untested)

Reply:
I should probably take a look at the URL spec.

The problem with your regex is that it would allow `';thing#`, for example